Assets/Scripts/DataTypes/Frequencies.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataTypes
+{
+    public class Frequencies
+    {
+        private int[] _frequencies { get; }
+        // counter for ticks since start
+        private int _ticks { get; set; }
+
+        public Frequencies(int[] frequencies)
+        {
+            _frequencies = frequencies;
+        }
+
+        public IEnumerable<int> CurrentActiveIndices()
+        {
+            var oldTicks = _ticks++;
+            return _frequencies.Where(frequency => oldTicks % frequency == 0).Select((frequency, index) => index);
```

Comment:
Wozu muss oldTicks zwischengespeichert werden?

Reply:
wegen dem inkrementieren

Reply:
Es wird doch aber nach dem Vergleichen inkrementiert...

Reply:
Soll es doch auch?

Reply:
Dann kann es doch direkt in der Where Abfrage aufgerufen werden, oder irre ich mich?

Reply:
Das in der Where Abfrage ist ne anonyme Lambda Funktion, die nen bool zurückliefert.
D.h. dass sie wie ne normale Funktion für jede frequency einmal aufgerufen wird.
Würde man dort also ticks inkrementieren, so würde dies für jede frequency 1 mal passieren.

Reply:
Jetzt hat's klick gemacht